### DISCUSSION TITLE
Close : Suggestion for bar scanner with ZBar project. Physical scanner.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#176] [Ideas]
I return to the request: 216. Bar scanner. (TBD) of the ToDoList because it is for me an essential option to adopt BiblioteQ for our library.
It will surely impact the request 276. Improve reservation processes. (TBD).

while searching a little I found the project http://zbar.sourceforge.net/  -> https://github.com/ZBar/ZBar -> https://git.linuxtv.org/zbar.git/ (0.23) 
after some tests it recognizes my barcodes taken with a smartphone and transfered to the computer

the project has a version with an api for Qt4
http://zbar.sourceforge.net/api/classzbar_1_1QZBar.html

I attach 2 examples of OK scans and one that I voluntarily blurred to see if the zbarimg command could recognize it
it's not the case because it detects 2 of them.

I hope that this will help you in this request suggestion

[test-barrecode.zip](https://github.com/textbrowser/biblioteq/files/7862329/test-barrecode.zip)

**Comment by textbrowser** [comment]
I am now stressed.

**Reply by meteos77** [reply]
no stress for you and make you pleasure it is the essential!
considering the time you spend I hope that the motivation is still there :-)

moreover there is time before our change to make a nice feature with the management of barcodes expected for a long time by all users who manage their documents with barcodes.

**Comment by textbrowser** [comment]
I am a strange person. My motivation grows as I age. So it is not motivation that stresses me. It is the adoption of hobbies with real-world deadlines and expectations.

**Reply by meteos77** [reply]
I'm the opposite, I zap, I'm passionate about something for a while but not on the long run, I'm interested in something else and I pick up
 (20 years of development of BiblioteQ it's project follow-up)

so for you no stress, it is only a discussion, I try to propose you a solution to advance on the subject. it can be bad, or good has you which develop to see.

if BiblioteQ is not ready, I will have other choices to present in May

1 - we wait with the old software (hoping it will last) that BiblioteQ proposes the function

2 - we buy a ready-made software (my motivation will drop instantly :-)

**Comment by meteos77** [comment]
I tell you my arrival on the subject of barcodes yesterday and today.

I am interested in the haiku bone (opensource successor of Beos)
I compiled **BiblioteQ on the bone with success** but yaz was missing for the Z3950.
I tried to compile yaz on haiku but some dependencies are missing for the compilation. I'm looking for possible librairy to compile and I find by chance zbar
I do some tests and suggest it to you :-)

**Reply by textbrowser** [reply]
YAZ is optional in BQ. :P

**Comment by textbrowser** [comment]
So you can compile it without YAZ and the project files are clear about doing so.

**Reply by meteos77** [reply]
it does indeed compile perfectly but I am looking for this feature

**Comment by textbrowser** [comment]
See BIBLIOTEQ_LINKED_WITH_YAZ.

**Comment by textbrowser** [comment]
I'm familiar with the project as someone told me they ported Dooble to it.

**Reply by meteos77** [reply]
I just installed it, but no French version :-(
I'll surely take care of it, especially if it's on haiku :-)

**Reply by textbrowser** [reply]
Which version of Dooble?

**Reply by textbrowser** [reply]
If it is 1.x, that is deprecated.

**Reply by meteos77** [reply]
1.56f

**Reply by meteos77** [reply]
I will test the compilation :-)
result :  some messages about dictionaries but it works fine with today's git clone version

**Reply by meteos77** [reply]
indeed a lot of version since

**Comment by textbrowser** [comment]
Why not SRU or Open Library?

**Reply by meteos77** [reply]
because my test book was not known about SRU and OpenLibrary and so I wanted to try byZ3950 ... which didn't work and so I switched to the yaz compilation which didn't work ... 
my operation is not turned towards the result but rather on the discovery.
I just tested another book and it's ok for SRU and openLibrary and cover

**Reply by meteos77** [reply]
it's great to be able to compile the software on any system that has qt (they just put qt6.2)

**Comment by textbrowser** [comment]
Have you tested yaz-client?

**Reply by meteos77** [reply]
there is no installation package
it is mandatory compilation and it blocks on a configuration dependency
getaddrinfo

**Comment by textbrowser** [comment]
We don't need to integrate ZBar. BQ works with any recognized bar scanner.
